core/src/main/java/com/datastax/oss/driver/internal/core/addresstranslation/SubnetAddressTranslator.java

```diff
@@ -0,0 +1,235 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This translator returns the proxy address of the private subnet containing the Cassandra node IP,
+ * or default address if no matching subnets, or passes through the original node address if no
+ * default configured.
+ *
+ * <p>The translator can be used for scenarios when all nodes are behind some kind of proxy, and
+ * that proxy is different for nodes located in different subnets (eg. when Cassandra is deployed in
+ * multiple datacenters/regions). One can use this, for example, for Cassandra on Kubernetes with
+ * different Cassandra datacenters deployed to different Kubernetes clusters.
+ */
+public class SubnetAddressTranslator implements AddressTranslator {
+  private static final Logger LOG = LoggerFactory.getLogger(SubnetAddressTranslator.class);
+
+  /**
+   * A map of Cassandra node subnets (CIDR notations) to target addresses, for example (note quoted
+   * keys):
+   *
+   * <pre>
+   * advanced.address-translator.subnet-addresses {
+   *   "100.64.0.0/15" = "cassandra.datacenter1.com:9042"
+   *   "100.66.0.0/15" = "cassandra.datacenter2.com:9042"
+   *   # IPv6 example:
+   *   # "::ffff:6440:0/111" = "cassandra.datacenter1.com:9042"
+   *   # "::ffff:6442:0/111" = "cassandra.datacenter2.com:9042"
+   * }
+   * </pre>
+   *
+   * Note: subnets must be represented as prefix blocks, see {@link
+   * inet.ipaddr.Address#isPrefixBlock()}.
+   */
+  public static final String ADDRESS_TRANSLATOR_SUBNET_ADDRESSES =
+      "advanced.address-translator.subnet-addresses";
+
+  /**
+   * A default address to fallback to if Cassandra node IP isn't contained in any of the configured
+   * subnets.
+   */
+  public static final String ADDRESS_TRANSLATOR_DEFAULT_ADDRESS =
+      "advanced.address-translator.default-address";
+
+  /**
+   * Whether to resolve the addresses on initialization (if true) or on each node (re-)connection
+   * (if false). Defaults to false.
+   */
+  public static final String ADDRESS_TRANSLATOR_RESOLVE_ADDRESSES =
+      "advanced.address-translator.resolve-addresses";
+
+  public static DriverOption ADDRESS_TRANSLATOR_SUBNET_ADDRESSES_OPTION =
+      new DriverOption() {
+        @NonNull
+        @Override
+        public String getPath() {
+          return ADDRESS_TRANSLATOR_SUBNET_ADDRESSES;
+        }
+      };
+
+  public static DriverOption ADDRESS_TRANSLATOR_DEFAULT_ADDRESS_OPTION =
+      new DriverOption() {
+        @NonNull
+        @Override
+        public String getPath() {
+          return ADDRESS_TRANSLATOR_DEFAULT_ADDRESS;
+        }
+      };
+
+  public static DriverOption ADDRESS_TRANSLATOR_RESOLVE_ADDRESSES_OPTION =
+      new DriverOption() {
+        @NonNull
+        @Override
+        public String getPath() {
+          return ADDRESS_TRANSLATOR_RESOLVE_ADDRESSES;
+        }
+      };
+
+  private final List<SubnetAddress> subnetAddresses;
+  private final Optional<InetSocketAddress> defaultAddress;
```

Comment:
mind adding this?
```suggestion
  
  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
  private final Optional<InetSocketAddress> defaultAddress;
```

It bothers me that IntelliJ doesn't look this as I think its completely fine, from [the docs](https://www.jetbrains.com/help/inspectopedia/OptionalUsedAsFieldOrParameterType.html) it looks like the main concern is around `Optional` not being `Serializable`, but that doesn't apply here and  I like reasoning about `Optional` types than checking for null.